### DISCUSSION TITLE
인프라 작업 목록

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#394] [이야기해봐요]
- [ ] CI
    - Actions 스크립트 작성
  - [ ] Production 배포
    - EC2 인스턴스 생성
    - RDS
    - Spring Boot
    - Nginx
      - HTTPS
    - Promtail
  - [ ] CD
    - Actions 스크립트 작성
  - [ ] 도메인 변경
    - 프로덕션 정적 파일 서버: code-zap.com
      - 개발 정적 파일 서버: dev.code-zap.com
      - 프로덕션 API 서버: api.code-zap.com
    - 개발 API 서버
      - dev.api.code-zap.com
      - beta.code-zap.com
- [ ] Develop 환경 리팩토링
  - EC2 인스턴스 scale-down
    - medium → small
  - 배포 환경 도커화
    - Spring Boot
      - `code-zap*.jar`
      - `application*.yml`
    - Nginx
      - HTTPS 적용
    - Promtail

**Comment by kyum-q** [comment]
## CI/CD 전에 읽어보면 구축 시 더 도움이 될 것 같아요 

- [CI 구축 방법](https://github.com/woowacourse-teams/2024-code-zap/wiki/CI-%ED%8C%8C%EC%9D%B4%ED%94%84%EB%9D%BC%EC%9D%B8-%EA%B5%AC%EC%B6%95)
- [CD 구축 방법](https://github.com/woowacourse-teams/2024-code-zap/wiki/CD-%ED%8C%8C%EC%9D%B4%ED%94%84%EB%9D%BC%EC%9D%B8-%EA%B5%AC%EC%B6%95)
   - self-hosted-runner 설정 방법 포함

- [동작 프로세스](https://github.com/woowacourse-teams/2024-code-zap/wiki/%EB%B0%B1%EC%97%94%EB%93%9C-CI-CD-%EB%8F%99%EC%9E%91-%ED%94%84%EB%A1%9C%EC%84%B8%EC%8A%A4)
   - 추후 동작 프로세스 구조도 변경 필요 
- [초기 CI/CD 트러블 슈팅](https://github.com/woowacourse-teams/2024-code-zap/wiki/%EB%B0%B1%EC%97%94%EB%93%9C-CI-CD-%ED%8A%B8%EB%9F%AC%EB%B8%94%EC%8A%88%ED%8C%85)
